File: src/Application Specific/Classes/Entity Collections/Spares/ProductTransaction/ProductTransaction.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace ProductTransactions

Public Class ProductTransaction

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "ProductTransaction Properties"

      
Private _ProductTransactionID As Integer = 0
Public Readonly Property ProductTransactionID() As Integer
	Get 
		Return _ProductTransactionID
	End Get	
End Property	
Public Writeonly Property SetProductTransactionID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ProductTransactionID", Value)
	End Set
End Property


Private _ProductID As Integer = 0
Public Readonly Property ProductID() As Integer
	Get 
		Return _ProductID
	End Get	
End Property	
Public Writeonly Property SetProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ProductID", Value)
	End Set
End Property


Private _Amount As Integer = 0
Public Readonly Property Amount() As Integer
	Get 
		Return _Amount
	End Get	
End Property	
Public Writeonly Property SetAmount() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Amount", Value)
	End Set
End Property


Private _TransactionDate As Datetime = Nothing
Public Property TransactionDate() As Datetime
	Get 
		Return _TransactionDate
	End Get
	Set (ByVal Value As Datetime)
		_TransactionDate = Value
	End Set
End Property	


Private _UserID As Integer = 0
Public Readonly Property UserID() As Integer
	Get 
		Return _UserID
	End Get	
End Property	
Public Writeonly Property SetUserID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_UserID", Value)
	End Set
End Property


Private _TransactionTypeID As Integer = 0
Public Readonly Property TransactionTypeID() As Integer
	Get 
		Return _TransactionTypeID
	End Get	
End Property	
Public Writeonly Property SetTransactionTypeID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_TransactionTypeID", Value)
	End Set
End Property


Private _LocationID As Integer = 0
Public Readonly Property LocationID() As Integer
	Get 
		Return _LocationID
	End Get	
End Property	
Public Writeonly Property SetLocationID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_LocationID", Value)
	End Set
End Property


Private _OrderProductID As Integer = 0
Public Readonly Property OrderProductID() As Integer
	Get 
		Return _OrderProductID
	End Get	
End Property	
Public Writeonly Property SetOrderProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderProductID", Value)
	End Set
End Property


Private _OrderLocationProductID As Integer = 0
Public Readonly Property OrderLocationProductID() As Integer
	Get 
		Return _OrderLocationProductID
	End Get	
End Property	
Public Writeonly Property SetOrderLocationProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderLocationProductID", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/ProductTransaction/ProductTransaction.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace ProductTransactions
    {
        public class ProductTransaction
        {
            private DataTypeValidator _dataTypeValidator;

            public ProductTransaction()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _ProductTransactionID = 0;

            public int ProductTransactionID
            {
                get
                {
                    return _ProductTransactionID;
                }
            }

            public object SetProductTransactionID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ProductTransactionID", value);
                }
            }

            private int _ProductID = 0;

            public int ProductID
            {
                get
                {
                    return _ProductID;
                }
            }

            public object SetProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ProductID", value);
                }
            }

            private int _Amount = 0;

            public int Amount
            {
                get
                {
                    return _Amount;
                }
            }

            public object SetAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Amount", value);
                }
            }

            private DateTime _TransactionDate = default;

            public DateTime TransactionDate
            {
                get
                {
                    return _TransactionDate;
                }

                set
                {
                    _TransactionDate = value;
                }
            }

            private int _UserID = 0;

            public int UserID
            {
                get
                {
                    return _UserID;
                }
            }

            public object SetUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_UserID", value);
                }
            }

            private int _TransactionTypeID = 0;

            public int TransactionTypeID
            {
                get
                {
                    return _TransactionTypeID;
                }
            }

            public object SetTransactionTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TransactionTypeID", value);
                }
            }

            private int _LocationID = 0;

            public int LocationID
            {
                get
                {
                    return _LocationID;
                }
            }

            public object SetLocationID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LocationID", value);
                }
            }

            private int _OrderProductID = 0;

            public int OrderProductID
            {
                get
                {
                    return _OrderProductID;
                }
            }

            public object SetOrderProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderProductID", value);
                }
            }

            private int _OrderLocationProductID = 0;

            public int OrderLocationProductID
            {
                get
                {
                    return _OrderLocationProductID;
                }
            }

            public object SetOrderLocationProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderLocationProductID", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}